sdk/nodejs/src/cloudClient.ts

```diff
@@ -22,13 +21,34 @@ const castIncomingUser = (user: DVCUser) => {
     return user
 }
 
-const throwIfUserError = (err: AxiosError) => {
-    const data = err.response?.data as { message?: string }
-    const code = err.response?.status || 0
+/**
+ * Handle response errors from API
+ * Suppress 5xx errors and log them instead while returning defaults to the caller
+ * Throw 4XX errors back to the caller to notify of invalid SDK usage
+ * Special handling of 404 error, which indicates "variable not found"
+ * @param err
+ */
+const throwIfUserError = (err: unknown) => {
+    if (err instanceof ResponseError) {
+        const code = err.status
 
-    if (code !== 404 && code < 500) {
-        throw new Error(`DevCycle request failed with status ${code}. ${data.message || ''}`)
+        // throw the error if it indicates there was user error in this call
+        // (e.g. invalid auth credentials or bad user data)
+        if (code !== 404 && code < 500) {
+            throw new Error(`DevCycle request failed with status ${code}. ${err.message || ''}`)
+        }
+
+        // Catch non-4xx errors so we can log them instead
+        return
     }
+
+    if (err instanceof SyntaxError) {
+        // JSON parse error, log instead of throwing
```

Comment:
where does it get logged?

Reply:
in the catch statements that are calling this. This method was a new thing we added to throw on 4xx errors, before this we just didn't have this method so the errors would always get caught and logged